app/managers/performance_metrics_manager/index.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {getTimeSinceStartup} from 'react-native-startup-time';
+
+import Batcher from './performance_metrics_batcher';
+
+type Target = 'HOME' | 'CHANNEL' | 'THREAD' | undefined;
+type MetricName = 'mobile_channel_switch' |
+    'mobile_team_switch';
+
+class PerformanceMetricsManager {
+    private startTimes: {[metricName: string]: undefined | number} = {};
+    private target: Target;
+    private batchers: {[serverUrl: string]: Batcher} = {};
+
+    private ensureBatcher(serverUrl: string) {
+        if (this.batchers[serverUrl]) {
+            return this.batchers[serverUrl];
+        }
+
+        this.batchers[serverUrl] = new Batcher(serverUrl);
+        return this.batchers[serverUrl];
+    }
+
+    public setLoadTarget(target: Target) {
+        this.target = target;
+    }
+
+    public finishLoad(location: Target, serverUrl: string) {
+        if (this.target !== location) {
+            return;
+        }
+
+        getTimeSinceStartup().then((measure) => {
+            this.ensureBatcher(serverUrl).addToBatch({
+                metric: 'mobile_load',
+                value: measure / 1000,
+                timestamp: Date.now(),
+            });
+        });
+    }
+
+    public startMetric(metricName: MetricName) {
+        this.startTimes[metricName] = global.performance.now();
+    }
+
+    public endMetric(metricName: MetricName, serverUrl: string) {
+        const startTime = this.startTimes[metricName];
+        if (startTime === undefined) {
+            return;
+        }
+
+        const dateNow = Date.now();
+        const performanceNow = global.performance.now();
+        const performanceNowSkew = dateNow - performanceNow;
+
+        this.ensureBatcher(serverUrl).addToBatch({
+            metric: metricName,
+            value: (performanceNow - startTime) / 1000,
+            timestamp: Math.round(startTime + performanceNowSkew),
```

Comment:
Is `performance.timeOrigin` not available for use here? I remember you mentioned that the performance API is only partially implemented on RN, but I don't know if that one is included

Reply:
I thought I did, and dismissed it because it wasn't maintained for a long time, but I see the last release was March, so I may have mixed different libraries in my mind.

I will re-assess whether this library makes sense for what we want to do.